.github/workflows/scripts/check_docstrings.py

```
@@ -0,0 +1,773 @@
+
+    Returns:
+        list: List of violations found in the file, with details about
+            the issue and corrective action.
+
+    """
+    # Initialize key variables
+    violations = []
+
+    # Read the file for processing
+    try:
+        with open(file_path, "r", encoding="utf-8") as fh_:
```

Ruff (0.11.9)

Unnecessary mode argument

Remove mode argument

(UP015)